src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1090,61 +1087,117 @@ private static NamedTypeSymbol GetDelegateType(BoundExpression expr)
                     ? receiver.Type
                     : this.ContainingType;
                 method = new ErrorMethodSymbol(methodContainer, returnType, name);
-                args = BuildArgumentsForErrorRecovery(analyzedArguments);
             }
 
+            // until we've considered the extension methods, we don't rewrite the arguments.
+            args = extensionMethodsOfSameViabilityAreAvailable
+                ? analyzedArguments.Arguments.ToImmutable()
+                : BuildArgumentsForErrorRecovery(analyzedArguments, methods);
+
             var argNames = analyzedArguments.GetNames();
             var argRefKinds = analyzedArguments.RefKinds.ToImmutableOrNull();
             return BoundCall.ErrorCall(node, receiver, method, args, argNames, argRefKinds, isDelegate, invokedAsExtensionMethod: invokedAsExtensionMethod, originalMethods: methods, resultKind: resultKind);
         }
 
-        private static ImmutableArray<BoundExpression> BuildArgumentsForErrorRecovery(AnalyzedArguments analyzedArguments, ImmutableArray<ParameterSymbol> parameters)
+        private bool IsUnboundGeneric(MethodSymbol method)
```

Comment:
I'd prefer not to make it generally available as an extension until the need has been proven. When it is needed elsewhere, that will be a good time to move it. There would not be much to say in the documentation that isn't said in the name, and if that isn't clear by reading the one line of its implementation.

The terms "unbound" and "constructed" are opposites in the spec. The term "unconstructed" does not appear in the ECMA specification at all, so I'm good with the name here.